### DISCUSSION TITLE
Sync Error can not be removed (Mountain Duck)

### DISCUSSION THREAD

**Discussion by andymachs** [iterate-ch/cyberduck#16356] [Q&A]
I Mounted WebDAV drive as a local disk via Mountain Duck, My WebDAV drive has Read-only access. no write.
my settings on Mountain Duck as follows:
Connect mode: Smart Synchronization
Enable buffering: On
Cache: default

When I opened a .xlsx or .docx file, a temporary file will generate, Temporary files are deleted as soon as the application closes the file after reading, At this point, Mountain Duck will indicate that there is a synchronization error. because of Read-only access of WebDAV. I also can not delete the temporary files manually, because temporary files were missing, I can not find them, but the Sync Error still existed,
how to solve this issue?

TKS!

**Comment by andymachs** [comment]
I don't think these temporary files should be treated as synchronized files because temporary files are deleted as soon as the application closes the file after reading, But Mountain Duck should have the ability to recognize temporary files and ignore synchronizing them.

**Reply by dkocher** [reply]
The default [setting](https://docs.mountainduck.io/tutorials/hidden_properties/) that matches temporary files that are not synched is

```
fs.filenames.temporary.regexp=\\.nfs\\..* \\.VolumeIcon.icns \\.DS_Store .*~$ .*~\\..* \\._~\\$.* \\.~(?!lock).*[^#] .*\\.tmp ~.*\\.tmp .*~.*\\.TMP .*\\.swap .*\\.swp Word\\sWork\\sFile.*\\.tmp \\._Word\\sWork\\sFile.*\\.tmp (.*)\\.sb-[\\w]{8}-[\\w]{6} \\._.*\\.(doc|docx)\\.sb-.* \\.TemporaryItems \\.dat\\.nosync.* TestFile[\\d]+-?[\\d]+$ .*~[\\w]{8}-[\\w]{4}-[\\w]{4}-[\\w]{4}-[\\w]{12}$ Adobe\\ Photoshop\\ [\\d]+$ DBTmp.* .*\\.lck [0-9A-Z]+\\.tmp .*\\.idlk Desktop\\.ini desktop\\.ini Thumbs\\.db \\(A\\ Document\\ Being\\ Saved\\ By\\ .*\\) ai[\\d]{10}.* AI\\ Temp.* .*\\.crdownload .*\\.part .*\\.onetmp \\.goutputstream-.* Adobe\\ Premiere\\ Pro\\ Auto-Save \\.[\\w]{2}\\.[\\w]{5}(\\.nosync)? \\.apdisk .*\\.lrdata .*\\.lst\\..*\\.[0-9]* .*~[\\w]{8}-[\\w]{4}-[\\w]{4}-[\\w]{4}-[\\w]{12}.prlock .*\\.[\\d]{4,5}\\.[\\d]{7,9}\\.aep
```

**Reply by andymachs** [reply]
Maybe in Smart Synchronization connect mode, Mountain Duck is not working or has bugs.

**Comment by dkocher** [comment]
It should be possible to delete the temporary local only file that has the error badge applied.

**Reply by andymachs** [reply]
Temporary files are deleted as soon as the application closes the file after reading,  so I can not delete it, but Mountain Duck still thinks it exists. so Sync Error keeping.

**Comment by andymachs** [comment]
In any case, as soon as the temporary files in the current folder are deleted, Mountain Duck should automatically cancel the Sync Error flag.
because the temporary files alreay do not exist, Equivalent manual deletion，There are not temporary files to Sync.

**Comment by andymachs** [comment]
Mountain Duck should focus on the current folder, keep Syncing all files, Whenever files are deleted, Mountain Duck should know about it instantly, it wouldn't have to synchronize them, and it shouldn't report an Sync error.

**Comment by andymachs** [comment]
If any file is deleted by applications or person on local or cloud storages, Mountain Duck should know them immediately and give up Sync, and also Mountain Duck can indicate that the file is deleted and provide a notice to person, In this case, Mountain Duck must not provide Sync Error flag.

**Comment by andymachs** [comment]
In Smart Synchronization connect mode and connect WebDAV ( Read-only), I open .xlsx, .docx, .pptx files, ~$** files are produced immediately, and then Mountain Duck always indicates Sync Error, even I can't delete the ~$**files. it still exists in current folder. Also the ~$** files do not disappear automatically.

**Reply by dkocher** [reply]
The lock owner files from Microsoft Office prefixed with `~ $` are synced when you have _Preferences → Sync… → Lock Files_ enabled to support [file locking](https://docs.mountainduck.io/mountainduck/locking/#pseudo-locking-for-protocols-with-no-native-lock-support-using-lock-owner-files).

**Reply by andymachs** [reply]
I try it, Using File locking can not solve the issue, it still indicate Sync Error.

**Comment by andymachs** [comment]
As long as the cloud storages has read-only access, 100% of Sync Errors will occur in Smart Synchronization connect mode.
even this Sync Error can not be removed, very worry about that. This problem is caused by temporary files, please find a solution to deal with temporary files.

**Comment by andymachs** [comment]
Open documents by some apps, it will produce some temporary files in current folder, I think Mountain Duck should deal with after close the apps, Once apps are closed, the temporary files will be delete automatically. if temporary files are deleted automatically, Mountain Duck do nothing.

**Comment by andymachs** [comment]
In Smart Synchronization connect mode, Maybe temporary files should not copy to the Cache. I failed to delete them, as if delete them, they appear soon.

**Reply by dkocher** [reply]
Could you [enable debug log output](https://docs.mountainduck.io/mountainduck/support/#logging-output) and send the log file after reproducing the problem to our [support team](mailto:support@mountainduck.io).

**Reply by andymachs** [reply]
![2024-09-20_213817](https://github.com/user-attachments/assets/996245ec-f145-4417-98b3-775e33201c00)

Please refer to screenshot,
Connect mode: Smart Synchronization
Enable Buffering: On
Locking files: Off
WebDAV: Read-only

**Comment by andymachs** [comment]
I see Documents of Mountain Duck:
_In Smart Synchronization connect mode, the buffer contents is moved from the temporary folder to the local cache and the status of the file changes to Up to Date._
=> Buffer contents will be deleted when quit the app, When quit the app, these buffer files are no longer useful, Why move them from the temporary folder to the local cache?
